service/request_enrollment.go

```diff
@@ -24,17 +37,41 @@ type enrollmentResponse struct {
 
 func decodeGRPCEnrollmentRequest(_ context.Context, grpcReq interface{}) (interface{}, error) {
 	req := grpcReq.(*pb.EnrollmentRequest)
+	enrollDetails := req.GetEnrollmentDetails()
 	return enrollmentRequest{
```

Comment:
similarly to the encode below, i'd like to see you fill out an `EnrollmentDetails` struct out and pass it in on line 44 instead

Reply:
what's the difference?

Reply:
just consistency - also i should have marked my comment as a nit